server/src/logic.js

```diff
@@ -234,6 +235,55 @@ export const timeSegmentHandler = {
   user(timesegment) {
     return timesegment.getUser();
   },
+  createTimeSegment(_, args, ctx) {
+    const { scheduleId, status, startTime, endTime } = args.timeSegment;
+    return getAuthenticatedUser(ctx)
+      .then(user =>
+        Schedule.findById(scheduleId).then((schedule) => {
+          if (!schedule) {
+            return Promise.reject(Error('Invalid schedule!'));
+          }
+          return Creators.timeSegment({
+            schedule,
+            status,
+            startTime,
+            endTime,
+            user,
+          });
+        }),
+      );
+  },
+  removeTimeSegment(_, args, ctx) {
+    const { segmentId } = args.timeSegment;
+    return getAuthenticatedUser(ctx)
+      .then(() =>
+        TimeSegment.findById(segmentId).then((segment) => {
+          if (!segment) {
+            return Promise.reject(Error('Invalid segment!'));
+          }
+          return segment.destroy().then((rows) => {
+            if (rows) { return true; }
+            return false;
+          });
+        }),
+      );
+  },
+  editTimeSegment(_, args, ctx) {
```

Comment:
prefer "update" to "edit" but dw for now

Reply:
Done